### ISSUE TITLE
Field Filter does not show Dropdown list

### ISSUE THREAD

**Issue by lucasloami** [metabase/metabase#7369]
Hi, I’m using Metabase v0.28.5 and I created questions in two ways: custom option and native query.

When I used the first option and added filters to my dashboard, they appear as a dropdown list. However, when I used the second option to create a similar question, the filter appears as a autocomplete, which is not the expected behavior, considering that the filter has only three possible options. The images below illustrate my point:

1. **Question created using custom option**: “Gênero” stands for “Gender” in Pt-BR.

![field filter gender - dropdown](https://user-images.githubusercontent.com/2198270/38941224-775bf0c6-4302-11e8-9aad-a93cbfce1b6a.png)

2. **Question created using native query**:

![field filter gender - autocomplete](https://user-images.githubusercontent.com/2198270/38941362-c4d26916-4302-11e8-8ecf-c06dc288b8b1.png)

Why is this happening? Both field filters refer to the same column in the database, so both should be presented as a dropdownlist.

This is the query I created:

```sql
SELECT 
    cluster,
    sum(monetary) as monetary_sum
FROM [TABLE_NAME]
WHERE
    1=1
    AND {{gender}}
GROUP BY cluster
ORDER BY cluster ASC
```

* Browsers: Google Chrome and Firefox
* Operating System: Windows and Ubuntu 17.10
* Metabase version: 0.28.5
* Metabase hosting environment: local docker instalation
* Metabase internal database: PostgreSQL

**Comment by lucasloami** [comment]
Hi, @hagen , in my Data Model this column is defined as "Category". I didn't find this _drop down list_ option you mentioned. I searched for it in the following versions and I was not able to find it: **v0.28.5**, **v0.28.6**, **v0.29.0-RC1**

**Comment by petar777** [comment]
I am experiencing the same problem in the last version of metabase.

**Comment by mazameli** [comment]
I couldn't reproduce this on either master or on v0.29-RC1. 

Master:
<img width="1294" alt="screen shot 2018-04-19 at 10 30 35 am" src="https://user-images.githubusercontent.com/2223916/39009099-819c29d0-43bf-11e8-8cd4-5baba362ecde.png">
<img width="1294" alt="screen shot 2018-04-19 at 10 30 38 am" src="https://user-images.githubusercontent.com/2223916/39009097-81800692-43bf-11e8-9047-e4bab5125af7.png">

RC1:

![screen shot 2018-04-19 at 10 37 42 am](https://user-images.githubusercontent.com/2223916/39009145-b4956f86-43bf-11e8-84e0-898ea01a6ad1.png)
![screen shot 2018-04-19 at 10 37 26 am](https://user-images.githubusercontent.com/2223916/39009146-b4b34024-43bf-11e8-8cdf-2b9af4c708d3.png)

@lucasloami to check that setting that @hagen mentioned, go to the Data Model in the Admin Panel, select the table this field is in, and then click the gear icon on the right hand side of that field to open the field settings for it. You'll see a filter options section. Can you tell me what it's set to? We've intermittently seen a problem where that setting isn't getting set correctly.

**Comment by hagen** [comment]
@lucasloami 
Against the field, where you pick 'Category' as the special type, there's a little gear icon.
![image](https://user-images.githubusercontent.com/5161952/39023441-fe6fb382-447d-11e8-96c7-6edd8b26fb53.png)

Click that, and go down to the section "Filtering on this field"
![image](https://user-images.githubusercontent.com/5161952/39023463-15ea7ce0-447e-11e8-8a9a-d65eceb177e6.png)

Switch to "A list of all values".

**Comment by lucasloami** [comment]
Hi, @mazameli and @hagen , thanks for your replies. 

I was able to solve the problem. I will describe below the things I discovered during the investigation of this problem. I'll do some assumptions on how Metabase scans fields; not sure if they're correct or not, but I believe it's worth mention.

1. **Metabase is impacted on how I modeled my database**: so, in order to have a quick deliverable in my project, I basically created a database with a denormalized schema (basically a big table with all the columns required without FK's). 

When I created questions using this schema and used "Gender" (defined as "Category" in data types) as a filter, I think Metabase scanned the whole table and identified that there were a lot of occurencies of "M", "F" and "I" without perform a `distinct` action in the field and it guessed the field would be a search box instead of a multiselect dropdown. *Would it be a improvement point for Metabase*?

2. **Solution 01: to change column's filter data type**: I was able to change the filter type as you described, and it solved my problem. It was the manual solution. 

3. **Solution 02: to change database modeling**: for performance reasons I modeled my database using a fact-dimension model and when I created the filters in the dashboard they were automatically guessed as multiselect dropdown.
